Source/ModelGroups/Mesh_Primitives.cs

```diff
@@ -1,193 +1,114 @@
+
+        public Mesh_Primitives(List<string> imageList)
         {
-            modelGroupName = ModelGroupName.Mesh_Primitives;
-            onlyBinaryProperties = false;
-            noPrerequisite = false;
-            Runtime.Image figureIndicesPrimitive0 = new Runtime.Image
-            {
-                Uri = imageList.Find(e => e.Contains("Indices_Primitive0"))
-            };
-            Runtime.Image figureIndicesPrimitive1 = new Runtime.Image
-            {
-                Uri = imageList.Find(e => e.Contains("Indices_Primitive1"))
-            };
+            UseFigure(imageList, "Indices_Primitive0");
+            UseFigure(imageList, "Indices_Primitive1");
 
-            usedFigures.Add(figureIndicesPrimitive0);
-            usedFigures.Add(figureIndicesPrimitive1);
+            // Track the common properties for use in the readme.
+            var baseColorFactorGreen = new Vector4(0.0f, 1.0f, 0.0f, 1.0f);
+            var baseColorFactorBlue = new Vector4(0.0f, 0.0f, 1.0f, 1.0f);
+            CommonProperties.Add(new Property(PropertyName.Material0WithBaseColorFactor, baseColorFactorGreen));
+            CommonProperties.Add(new Property(PropertyName.Material1WithBaseColorFactor, baseColorFactorBlue));
 
-            List<Vector3> primitive0Positions = new List<Vector3>()
-            {
-                new Vector3(-0.5f,-0.5f, 0.0f),
-                new Vector3( 0.5f, 0.5f, 0.0f),
-                new Vector3(-0.5f, 0.5f, 0.0f)
-            };
-            List<Vector3> primitive1Positions = new List<Vector3>()
+            Model CreateModel(Action<List<Property>, List<Runtime.MeshPrimitive>, Runtime.PbrMetallicRoughness, Runtime.PbrMetallicRoughness> setProperties)
             {
-                new Vector3(-0.5f,-0.5f, 0.0f),
-                new Vector3( 0.5f,-0.5f, 0.0f),
-                new Vector3( 0.5f, 0.5f, 0.0f)
-            };
-            List<int> primitiveTriangleIndices = new List<int>
-            {
-                0, 1, 2,
-            };
-            Runtime.MeshPrimitive primitive0Mesh = new Runtime.MeshPrimitive
-            {
-                Positions = primitive0Positions,
-                Indices = primitiveTriangleIndices,
-            };
-            Runtime.MeshPrimitive primitive1Mesh = new Runtime.MeshPrimitive
-            {
-                Positions = primitive1Positions,
-                Indices = primitiveTriangleIndices,
-            };
-            Vector4 colors0 = new Vector4(0.0f, 1.0f, 0.0f, 1.0f);
-            Vector4 colors1 = new Vector4(0.0f, 0.0f, 1.0f, 1.0f);
+                var properties = new List<Property>();
+                var meshPrimitives = MeshPrimitive.CreateMultiPrimitivePlane();
+                foreach (var primitive in meshPrimitives)
+                {
+                    primitive.TextureCoordSets = null;
+                }
+                meshPrimitives[0].Material = new Runtime.Material()
```

Comment:
This seems more complicated than it should be. Let's discuss offline.

Reply:
Create the materials in the helper functions instead of here.

Reply:
Fixed